### ISSUE TITLE
How big should a controlnet dataset be for training?

### ISSUE THREAD

**Issue by Atoli** [lllyasviel/ControlNet#569]
The training docs use a 50,000 base image dataset zipped.
The example involves something simple as circles, in a real training setting you might want to create a control for more complex things such as noses or eyes.
It would take months if not years if i tried to manually draw noses outlines for 50,000 images to use to train, without mention find 50,000 good photos.

Just how big should the average dataset be for a non-example controlnet training, realistically?

**Comment by geroldmeisinger** [comment]
I think the fun and creativity in machine learning comes exactly from finding a way to get this training dataset WITHOUT having to manually draw 50000 noses. Read the paper and look how they synthetized the scribble ControlNet for example: 1. hed-filter an image to get outlines of different strengths, 2. remove the thin ones, 3. redraw the thickest ones with scribble brush, done. that's incredible! Or read the Midas paper on how they synthetized images for depth maps: where do you get thousands of realistic 3D images from? well... 3D movies! that's ingenious! Or read the paper on how they created Instruct Pix2Pix: how do you get from a caption describing a image to caption describing a CHANGE of the image and then actually have an image that incorporates this exact meaningful change? they didn't manually inpaint thousands of images ;) there are already a few ControlNets for facial recognition and pose and they all made use of other machine learning models to predict and estimate faces and poses.

to your actual question. read my article https://civitai.com/articles/2078 and my report on a different edge ControlNet https://github.com/lllyasviel/ControlNet/discussions/318#discussioncomment-7202122 where I have a few numbers. it all depends on your
* batchsize: higher requires more images, but you get bettter quality
* learning rate: higher means you require less images, but your CN could explode
* epochs: you can train on the same images multiple times, but quality decreases
* concept: simpler condition images converge earlier
* number of images: if you look at my article, for the test ControlNet I still required 18000a4 images to see some effect
(you can find a few more numbers at the QA section below)

all in all, you will need thousand of images and you won't get around finding a way to automatically synthesize those images (plus captions!)

**Comment by Khaihuyennguyen** [comment]
> I think the fun and creativity in machine learning comes exactly from finding a way to get this training dataset WITHOUT having to manually draw 50000 noses. Read the paper and look how they synthetized the scribble ControlNet for example: 1. hed-filter an image to get outlines of different strengths, 2. remove the thin ones, 3. redraw the thickest ones with scribble brush, done. that's incredible! Or read the Midas paper on how they synthetized images for depth maps: where do you get thousands of realistic 3D images from? well... 3D movies! that's ingenious! Or read the paper on how they created Instruct Pix2Pix: how do you get from a caption describing a image to caption describing a CHANGE of the image and then actually have an image that incorporates this exact meaningful change? they didn't manually inpaint thousands of images ;) there are already a few ControlNets for facial recognition and pose and they all made use of other machine learning models to predict and estimate faces and poses.
> 
> to your actual question. read my article https://civitai.com/articles/2078 and my report on a different edge ControlNet [#318 (comment)](https://github.com/lllyasviel/ControlNet/discussions/318#discussioncomment-7202122) where I have a few numbers. it all depends on your
> 
> * batchsize: higher requires more images, but you get bettter quality
> * learning rate: higher means you require less images, but your CN could explode
> * epochs: you can train on the same images multiple times, but quality decreases
> * concept: simpler condition images converge earlier
> * number of images: if you look at my article, for the test ControlNet I still required 18000a4 images to see some effect
>   (you can find a few more numbers at the QA section below)
> 
> all in all, you will need thousand of images and you won't get around finding a way to automatically synthesize those images (plus captions!)

Hi, I am new to Computer Vision and still learning everyday and have a similar question.

I am going to extract frames from video(s) to train the model. However, the images will look familiar. For example, a plane flying from point A to point B in the video, background unchanged. Do you think it is the good idea to use all the frames for training.  If we need to choose images, how many is enough. 

Please noted that the frames are different but similar in background.

Appreciate your opinion on this.

**Comment by geroldmeisinger** [comment]
if you are going to train for video generation then you would need each frame. if you are just using the frames as an image source it's essentially a duplicate image which you want to avoid.

**Comment by Khaihuyennguyen** [comment]
> if you are going to train for video generation then you would need each frame. if you are just using the frames as an image source it's essentially a duplicate image which you want to avoid.

Will it make different if I added another drone, move the drone around?

My project is similar to the control net "circles" data set. Very simple drone with different background and positions

**Comment by geroldmeisinger** [comment]
please provide some example images. i don't know what you mean by "drone"

**Comment by Khaihuyennguyen** [comment]
> please provide some example images. i don't know what you mean by "drone."

Sorry for the confusing. 

Let's start with the fill50k dataset that ControlNet used as example. In the prompt, if you search for the sentence: **"corn silk circle with green background"** you will see there are two images have that same prompt: 48653.png, 18394.png. The only difference between two images is the size of the circle and position.

In my project, that is exactly what I need. I just want to generate simple object (drone, plane or bird) and simple background (clear sky,...) with different size and/or position. 

Therefore, my plan is that I extract a video which has the same background with different object size and position for training. My question is: for my purpose, will a similar background effect the result?
